crates/sui-framework/packages/sui-framework/sources/bridge/bridge.move

```diff
@@ -0,0 +1,363 @@
+    }
+
+    fun serialise_token_bridge_payload<T>(
+        sender: address,
+        target_chain: u8,
+        target_address: vector<u8>,
+        token: &Coin<T>
+    ): vector<u8> {
+        let payload = vector[];
+        vector::append(&mut payload, bcs::to_bytes(&address::to_bytes(sender)));
+        vector::push_back(&mut payload, target_chain);
+        vector::append(&mut payload, bcs::to_bytes(&target_address));
+        vector::append(&mut payload, bcs::to_bytes(&token_id<T>()));
+        vector::append(&mut payload, bcs::to_bytes(&balance::value(coin::balance(token))));
+        payload
+    }
+
+    fun deserialise_token_bridge_payload(message: vector<u8>): TokenBridgePayload {
+        let bcs = bcs::new(message);
+        let sender_address = bcs::peel_vec_u8(&mut bcs);
+        let target_chain = bcs::peel_u8(&mut bcs);
+        let target_address = bcs::peel_vec_u8(&mut bcs);
+        let token_type = bcs::peel_u8(&mut bcs);
+        let amount = bcs::peel_u64(&mut bcs);
+        TokenBridgePayload {
+            sender_address,
+            target_chain,
+            target_address,
+            token_type,
+            amount
+        }
+    }
+
+    fun deserialise_emergency_op_payload(message: vector<u8>): EmergencyOpPayload {
+        let bcs = bcs::new(message);
+        EmergencyOpPayload {
+            op_type: bcs::peel_u8(&mut bcs)
+        }
+    }
+
+    fun deserialise_message(message: vector<u8>): BridgeMessage {
+        let bcs = bcs::new(message);
+        BridgeMessage {
+            message_type: bcs::peel_u8(&mut bcs),
+            message_version: bcs::peel_u8(&mut bcs),
+            seq_num: bcs::peel_u64(&mut bcs),
+            source_chain: bcs::peel_u8(&mut bcs),
+            payload: bcs::into_remainder_bytes(bcs)
+        }
+    }
+
+    fun serialise_message(message: BridgeMessage): vector<u8> {
+        let BridgeMessage {
+            message_type,
+            message_version: version,
+            seq_num: bridge_seq_num,
+            source_chain,
+            payload
+        } = message;
+
+        let message = vector[];
+        vector::push_back(&mut message, message_type);
+        vector::push_back(&mut message, version);
+        vector::append(&mut message, bcs::to_bytes(&bridge_seq_num));
+        vector::push_back(&mut message, source_chain);
+        vector::append(&mut message, payload);
+        message
+    }
+
+    // Create bridge request to send token to other chain, the request will be in pending state until approved
+    public fun send_token<T>(
+        self: &mut Bridge,
+        target_chain: u8,
+        target_address: vector<u8>,
+        token: Coin<T>,
+        ctx: &mut TxContext
+    ) {
+        let bridge_seq_num = self.sequence_num;
+        self.sequence_num = self.sequence_num + 1;
+        // create bridge message
+        let payload = serialise_token_bridge_payload(tx_context::sender(ctx),target_chain, target_address, &token);
```

Comment:
Should we do some checking here that `target_chain` is not sui and is one of the supported chains?